### ISSUE TITLE
IMDB plugin crashing

### ISSUE THREAD

**Issue by rufftruffles** [Flexget/Flexget#3998]
### Expected behaviour:

To get list of movies from public watch list on imdb

### Steps to reproduce:

$ flexget execute --task movie-watchlist --no-cache --discover-now

### Actual behaviour:
Imdb plugin crashes:

```
2024-05-30 17:38:26 CRITICAL manager       movie-watchlist An unexpected crash has occurred. Writing crash report to /home/torrent/.flexget/crash_report.2024.05.30.173826122465.log. Please verify you are running the latest version of flexget by using "flexget -V" from CLI or by using version_checker plugin at https://flexget.com/Plugins/version_checker. You are currently using version 3.11.35
2024-05-30 17:38:26 WARNING  task          movie-watchlist Aborting task (plugin: imdb_watchlist) 
```

#### Config:
```
templates:
  anchors:
 ############################
# Torrent Client Credentials #
 ############################

    _transmission-settings: &transmission-settings
      host: localhost
      port: 12345
      username: XXXXX
      password: XXXXX

 ###########################
# Torrent site(s) credentials #
 ###########################

    _iptorrents_credentials: &iptorrents_credentials
      rss_key: XXXXX
      uid: XXXXX
      password: XXXXX

    _torrentleech_credentials: &torrentleech_credentials
      rss_key: XXXXX
      username: 'XXXXX'
      password: 'XXXXX'

 ###########################
# TV Series Lookup settings #
 ##########################

#  thetvdb_list:
#    username: XXXXX
#    account_id: XXXXX
#    api_key: XXXXX
#    strip_dates: yes

#  trakt-list: &trakt-list
#    trakt_list:
#      username: XXXXX
#      account: XXXXX
#      list: Download
#      type: shows
#      strip_dates: yes

 ######################
#  General Tv Settings  #
 ######################

  tv-settings:
    torrent_alive: 1
    trakt_lookup: yes
    tvmaze_lookup: yes
    series:
      - 'Shōgun':
          exact: yes
          alternate_name:
            - Shogun
            - Shogun 2024
    configure_series:
      from:
        thetvdb_list:
          username: XXXXX
          account_id: XXXXX
          api_key: XXXXX
          strip_dates: yes
#        <<: *trakt-list
      settings:
#        timeframe: 6 hours
#        quality: 2160p-1080p bluray|remux|webdl|webrip|hdtv
        target: 2160p bluray|remux|webdl|webrip|hdtv
#        quality: 2160p bluray|remux|webdl|webrip|hdtv dolbyvision|hdrplus|hdr|10bit|8bit
        quality: 1080p-2160p bluray|remux|webdl|webrip|hdtv
        timeframe: 6 hours
        propers: no
        identified_by: ep
#      from:
#        <<: *thetvdb_list

    notify:
      entries:
        title: "Downloading {{tvmaze_series_name}} {{tvmaze_series_year}}"
        message: |+
          {%if task in ["retreive_from_couchpotato","dynamic_imdb_actors"]%}*New movie added to queue*
          {%else%}*Download Started from task:
          *{{task|replace("_", "-")}}
          {%endif%}
          {% if series_name is defined -%}
          *Series*: {{series_name}}
          *Quality*: {{quality|d('')}}
          *Season/Episode*: {{trakt_ep_id}}
          *Rating*:  {{tvmaze_series_rating}}
          [Image]({{tvmaze_series_original_image|replace("_", "%5F")}})
          [Show page]({{tvmaze_series_url|replace("_", "%5F")}})
          {{quality|d('')}}
          {% else -%}
          {{title}}
          {%- endif -%}
        via:
          - telegram:
              bot_token: 'XXXXX'              
              parse_mode: markdown
              disable_previews: no
              recipients:
                - username: 'XXXXX'
              socks_proxy:
                url: socks5://XXXXX
                username: 'XXXXX'
                password: 'XXXXX'

 ########################
# General Movie Settings #
 ########################

  movie-settings:
    torrent_alive: 3
    imdb_lookup: yes
    tmdb_lookup: yes
    regexp:
      reject:
        - extended
        - will1869
        - life-static
    content_size:
      min: 2900
      max: 70000
    quality:
      - 2160p remux|bluray|webdl dolbyvision|hdrplus|hdr
#dolbyvision|hdrplus|hdr
#      - 1080p bluray|webdl
    list_match:
      from:
        - movie_list: Movies
    seen_movies: strict
    proper_movies: no
    notify:
      entries:
        title: "Downloading {{movie_name}} {{movie_year}}"
        message: |+
          {%if task in ["retreive_from_couchpotato","dynamic_imdb_actors"]%}*New movie added to queue*
          {%else%}*Download Started from task:
          *{{task|replace("_", "-")}}
          {%endif%}
          {% if series_name is defined -%}
          *{{series_name}}* - {{series_id}} - {{quality|d('')}}
          *{{tvmaze_episode_name|d(tvdb_ep_name)|d('')}}*
          [Image]({{tvmaze_series_original_image|replace("_", "%5F")}})
          [Show page]({{tvmaze_series_url|replace("_", "%5F")}})
          {% elif imdb_name is defined -%}
          *{{imdb_name}}* - ({{imdb_year}})
          {{quality|d('')}}
          {{imdb_score}}/10 - {{imdb_votes}} votes
          {{imdb_genres|join(', ')|title}}
          *Plot:* {{imdb_plot_outline}}
          [Image]({{tmdb_posters[0]|replace("_", "%5F")}})
          [Movie Page]({{imdb_url|d('')}})
          {% else -%}
          {{title}}
          {%- endif -%}
        via:
          - telegram:
              bot_token: 'XXXXX'
              parse_mode: markdown
              disable_previews: no
              recipients:
                - username: 'XXXXX'
              socks_proxy:
                url: socks5://XXXXX
                username: 'XXXXX'
                password: 'XXXXX'

tasks:

 ############################################
# Update Movie list from your IMDB watchlist #
 ############################################

  movie-watchlist:
    priority: 1
    imdb_lookup: yes
    tmdb_lookup: yes
    imdb_watchlist:
      user_id: XXXXX
      list: XXXXX
    accept_all: yes
    list_add:
      - movie_list: Movies
 ###################################################################################
# Search TV Episodes from IPtorrents RSS Feed and send to Transmission for download #
 ###################################################################################

  tv-rss:
    priority: 2
    template: tv-settings
    inputs:
      - rss:
#          url: https://rss24h.torrentleech.org/XXXXX
          url: https://iptorrents.com/torrents/rss?u=XXXXX;tp=XXXXX;99;22;download
          all_entries: no

    transmission:
      <<: *transmission-settings
      path: /transmission/Downloads/TV-SHOWS

 ##############################################################################
# Search Movies from IPtorrents RSS Feed and send to Transmission for download  #
 ##############################################################################

  movie-rss:
    priority: 3
    template: movie-settings
    rss:
      url: https://rss24h.torrentleech.org/XXXXX
#      url: https://iptorrents.com/torrents/rss?u=XXXXX;tp=XXXXX;101;48;20;100;38;download
      all_entries: no

    transmission:
      <<: *transmission-settings
      path: /transmission/Downloads/MOVIES

 ##############################################################################################################
# Search TV Episodes in IPtorrents / Torrentleech using search feature if episode not found in RSS  & Download #
 ##############################################################################################################
  tv-search:
    priority: 4
    template: tv-settings
    discover:
      what:
        - next_series_episodes:
            from_start: yes
      from:
        - torrentleech:
            <<: *torrentleech_credentials
            category: [32]
        - iptorrents:
            <<: *iptorrents_credentials
            category: [99, TV-Web-DL, TV-x264, 23]
    transmission:
      <<: *transmission-settings
      path: /transmission/Downloads/TV-SHOWS

 ###########################################################################################
# Search Movies in IPtorrents / Torrentleech using search feature if Movie not found in RSS #
 ###########################################################################################

  movie-search:
    priority: 5
    template: movie-settings
    discover:
      release_estimations:
        smart
#        optimistic: 90 days
      what:
        - movie_list: Movies
      from:
#            category: [Movie-4K, 101, 48, 20, 38, 100]
        - torrentleech:
            <<: *torrentleech_credentials
            category: [47]

        - iptorrents:
            <<: *iptorrents_credentials
            category: [101]
# 101, 20, 38, 100]

    transmission:
      <<: *transmission-settings
      path: /transmission/Downloads/MOVIES
```
  
#### Log:
<details>
 <summary>(click to expand)</summary>

```
2024-05-30 17:38:21 DEBUG    manager                       Figuring out config load paths
2024-05-30 17:38:21 DEBUG    manager                       Found config: /home/torrent/.flexget/config.yml
2024-05-30 17:38:21 DEBUG    manager                       Config file /home/torrent/.flexget/config.yml selected
2024-05-30 17:38:21 DEBUG    manager                       sys.defaultencoding: utf-8
2024-05-30 17:38:21 DEBUG    manager                       sys.getfilesystemencoding: utf-8
2024-05-30 17:38:21 DEBUG    manager                       flexget detected io encoding: utf-8
2024-05-30 17:38:21 DEBUG    manager                       os.path.supports_unicode_filenames: False
2024-05-30 17:38:21 DEBUG    manager                       Figuring out config load paths
2024-05-30 17:38:21 DEBUG    manager                       Found config: /home/torrent/.flexget/config.yml
2024-05-30 17:38:21 DEBUG    manager                       Config file /home/torrent/.flexget/config.yml selected
2024-05-30 17:38:21 DEBUG    manager                       sys.defaultencoding: utf-8
2024-05-30 17:38:21 DEBUG    manager                       sys.getfilesystemencoding: utf-8
2024-05-30 17:38:21 DEBUG    manager                       flexget detected io encoding: utf-8
2024-05-30 17:38:21 DEBUG    manager                       os.path.supports_unicode_filenames: False
2024-05-30 17:38:21 DEBUG    plugin                        Trying to load plugins from: ['/home/torrent/.flexget/plugins', '/home/torrent/flexget/lib/python3.11/site-packages/flexget/plugins']
2024-05-30 17:38:21 DEBUG    plugin                        Plugin `memusage` requires plugin `guppy3` to load.
2024-05-30 17:38:21 DEBUG    plugin                        Trying to load components from: ['/home/torrent/.flexget/components', '/home/torrent/flexget/lib/python3.11/site-packages/flexget/components']
2024-05-30 17:38:22 DEBUG    plugin                        Plugins took 1.04 seconds to load. 324 plugins in registry.
2024-05-30 17:38:22 DEBUG    manager                       Connecting to: sqlite:////home/torrent/.flexget/db-config.sqlite
2024-05-30 17:38:22 DEBUG    manager                       config_name: config
2024-05-30 17:38:22 DEBUG    manager                       config_base: /home/torrent/.flexget
2024-05-30 17:38:22 DEBUG    manager                       New config data loaded.
2024-05-30 17:38:22 DEBUG    schema                        current flexget version already exist in db 3.11.35
2024-05-30 17:38:22 DEBUG    parsing                       setting default movie parser to internal. (options: {'internal': <flexget.components.parsing.parsers.parser_internal.ParserInternal object at 0x7fea062846d0>, 'guessit': <flexget.components.parsing.parsers.parser_guessit.ParserGuessit object at 0x7fea06284810>})
2024-05-30 17:38:22 DEBUG    parsing                       setting default series parser to internal. (options: {'internal': <flexget.components.parsing.parsers.parser_internal.ParserInternal object at 0x7fea062846d0>, 'guessit': <flexget.components.parsing.parsers.parser_guessit.ParserGuessit object at 0x7fea06284810>})
2024-05-30 17:38:22 DEBUG    cron_env                      Encoding utf-8 stored
2024-05-30 17:38:22 DEBUG    util.simple_persistence                 setting key terminal_encoding value 'utf-8'
2024-05-30 17:38:22 DEBUG    task_queue                    task queue shutdown requested
2024-05-30 17:38:22 VERBOSE  task_queue                    There are 1 tasks to execute. Shutdown will commence when they have completed.
2024-05-30 17:38:22 DEBUG    task          movie-watchlist executing movie-watchlist
2024-05-30 17:38:22 DEBUG    input_cache   movie-watchlist cache name: imdb_watchlist_49b3cfb500fc4a36d981458eccef9732 (has: )
2024-05-30 17:38:22 VERBOSE  imdb_watchlist movie-watchlist Retrieving imdb list: ls090066867
2024-05-30 17:38:22 DEBUG    imdb_watchlist movie-watchlist Requesting: http://www.imdb.com/list/ls090066867 {'Accept-Language': 'en-us'}
2024-05-30 17:38:22 DEBUG    utils.requests movie-watchlist GETing URL http://www.imdb.com/list/ls090066867 with args () and kwargs {'params': {'view': 'detail', 'page': 1}, 'headers': {'Accept-Language': 'en-us'}, 'allow_redirects': True, 'timeout': 30}
2024-05-30 17:38:26 VERBOSE  imdb_watchlist movie-watchlist No movies were found in imdb list: ls090066867
2024-05-30 17:38:26 DEBUG    input_cache   movie-watchlist storing entries to cache imdb_watchlist_49b3cfb500fc4a36d981458eccef9732
2024-05-30 17:38:26 CRITICAL task          movie-watchlist BUG: Unhandled error in plugin imdb_watchlist: 'NoneType' object is not iterable
Traceback (most recent call last):

  File "/usr/lib/python3.11/threading.py", line 995, in _bootstrap
    self._bootstrap_inner()
    │    └ <function Thread._bootstrap_inner at 0x7fea0cb81260>
    └ <Thread(task_queue, started daemon 140643098298048)>
  File "/usr/lib/python3.11/threading.py", line 1038, in _bootstrap_inner
    self.run()
    │    └ <function Thread.run at 0x7fea0cb80f40>
    └ <Thread(task_queue, started daemon 140643098298048)>
  File "/usr/lib/python3.11/threading.py", line 975, in run
    self._target(*self._args, **self._kwargs)
    │    │        │    │        │    └ {}
    │    │        │    │        └ <Thread(task_queue, started daemon 140643098298048)>
    │    │        │    └ ()
    │    │        └ <Thread(task_queue, started daemon 140643098298048)>    │    └ <bound method TaskQueue.run of <flexget.task_queue.TaskQueue object at 0x7fea06113510>>
    └ <Thread(task_queue, started daemon 140643098298048)>
  File "/home/torrent/flexget/lib/python3.11/site-packages/flexget/task_queue.py", line 46, in run
    self.current_task.execute()
    │    │            └ <function Task.execute at 0x7fea0a262520>
    │    └ <flexget.task.Task object at 0x7fea061a5e10>
    └ <flexget.task_queue.TaskQueue object at 0x7fea06113510>
  File "/home/torrent/flexget/lib/python3.11/site-packages/flexget/task.py", line 87, in wrapper
    return func(self, *args, **kw)
           │    │      │       └ {}
           │    │      └ ()
           │    └ <flexget.task.Task object at 0x7fea061a5e10>
           └ <function Task.execute at 0x7fea0a262480>
  File "/home/torrent/flexget/lib/python3.11/site-packages/flexget/task.py", line 716, in execute
    self._execute()
    │    └ <function Task._execute at 0x7fea0a2623e0>
    └ <flexget.task.Task object at 0x7fea061a5e10>
  File "/home/torrent/flexget/lib/python3.11/site-packages/flexget/task.py", line 685, in _execute
    self.__run_task_phase(phase)
    │                     └ 'input'
    └ <flexget.task.Task object at 0x7fea061a5e10>
  File "/home/torrent/flexget/lib/python3.11/site-packages/flexget/task.py", line 514, in __run_task_phase
    response = self.__run_plugin(plugin, phase, args)
               │                 │       │      └ (<flexget.task.Task object at 0x7fea061a5e10>, {'user_id': 'ur77350165', 'list': 'ls090066867', 'force_language': 'en-us', 't...
               │                 │       └ 'input'
               │                 └ <PluginInfo(name=imdb_watchlist)>
               └ <flexget.task.Task object at 0x7fea061a5e10>
> File "/home/torrent/flexget/lib/python3.11/site-packages/flexget/task.py", line 547, in __run_plugin
    result = method(*args, **kwargs)
             │       │       └ {}
             │       └ (<flexget.task.Task object at 0x7fea061a5e10>, {'user_id': 'ur77350165', 'list': 'ls090066867', 'force_language': 'en-us', 't...
             └ <Event(name=plugin.imdb_watchlist.input,func=wrapped_func,priority=128)>
  File "/home/torrent/flexget/lib/python3.11/site-packages/flexget/event.py", line 21, in __call__
    return self.func(*args, **kwargs)
           │    │     │       └ {}
           │    │     └ (<flexget.task.Task object at 0x7fea061a5e10>, {'user_id': 'ur77350165', 'list': 'ls090066867', 'force_language': 'en-us', 't...
           │    └ <bound method cached.__call__.<locals>.wrapped_func of <flexget.components.imdb.imdb_watchlist.ImdbWatchlist object at 0x7fea...
           └ <Event(name=plugin.imdb_watchlist.input,func=wrapped_func,priority=128)>
  File "/home/torrent/flexget/lib/python3.11/site-packages/flexget/utils/cached_input.py", line 177, in wrapped_func
    cache = IterableCache(response, self.store_to_db if self.persist else None)
            │             │         │    │              │    └ datetime.timedelta(seconds=7200)
            │             │         │    │              └ <flexget.utils.cached_input.cached object at 0x7fea06761350>
            │             │         │    └ <function cached.store_to_db at 0x7fea07f9bba0>
            │             │         └ <flexget.utils.cached_input.cached object at 0x7fea06761350>
            │             └ None
            └ <class 'flexget.utils.cached_input.IterableCache'>
  File "/home/torrent/flexget/lib/python3.11/site-packages/flexget/utils/cached_input.py", line 225, in __init__
    self.iterable = iter(iterable)
    │                    └ None
    └ <flexget.utils.cached_input.IterableCache object at 0x7fea05f50dd0>

TypeError: 'NoneType' object is not iterable
Traceback (most recent call last):
  File "/home/torrent/flexget/lib/python3.11/site-packages/flexget/task.py", line 547, in __run_plugin
    result = method(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/torrent/flexget/lib/python3.11/site-packages/flexget/event.py", line 21, in __call__
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/torrent/flexget/lib/python3.11/site-packages/flexget/utils/cached_input.py", line 177, in wrapped_func
    cache = IterableCache(response, self.store_to_db if self.persist else None)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/torrent/flexget/lib/python3.11/site-packages/flexget/utils/cached_input.py", line 225, in __init__
    self.iterable = iter(iterable)
                    ^^^^^^^^^^^^^^
TypeError: 'NoneType' object is not iterable
```
</details>

### Additional information:

- FlexGet version: 3.11.35
- Python version: 3.11.2
- Installation method: pip
- Using daemon (yes/no): no
- OS and version: Debian 12
- Link to crash log:

**Comment by rufftruffles** [comment]
Hi @gazpachoking , could you please take a look?

Thanks

**Comment by rufftruffles** [comment]
Anyone?

**Comment by rufftruffles** [comment]
@BrutuZ would appreciate any help here please.

Thanks

**Comment by BrutuZ** [comment]
Duplicate of #3995 and as such same answer from me: I've never used IMDB Watchlists

**Comment by rufftruffles** [comment]
> Duplicate of #3995 and as such same answer from me: I've never used IMDB Watchlists

Thanks for the response @BrutuZ could you please check who handles the imdb plugin? I can't seem to find the author.

I'd like to ask him for help.

Thank you

**Comment by rufftruffles** [comment]
@liiight can we get your valuable suggestion here?  :)

**Comment by gazpachoking** [comment]
Sorry for my lack of response here, I have a newborn in the house and not much time.

Looks like imdb totally changed the format they are returning for the lists. I think probably the easiest way is to switch the plugin back to using the csv export that imdb provides. (I can't remember why we switched from that in the first place.)

For the watchlist we need to retrieve the list ID to do that before we can grab it, so it's probably a 2 step process. Grab https://www.imdb.com/user/urXXXXX/watchlist/ first and scrape the list ID (lsXXXXX format). Then it becomes the same as a specific list. Just grab the csv from https://www.imdb.com/list/lsXXXX/export

Anyone that wants to write up a PR would be much appreciated.

**Comment by chewy74** [comment]
I'm also having issues with imdb_list plugin. It's giving me 404 errors and not adding new entries. 
![Screenshot_20240627-203222_crDroid_Home](https://github.com/Flexget/Flexget/assets/7645787/470d3611-141b-41a4-a35e-1f5ef3b2b297)

**Comment by josh** [comment]
I believe the old CSV export URLs went away a few days ago. It seems IMDB now has a new [exports page](https://www.imdb.com/exports/?ref_=wl), but requires them to be explicitly queued up. I'm wondering, but not sure if that process can be automated.

**Comment by tavicu** [comment]
Any update regarding this issue?

**Comment by rufftruffles** [comment]
Any update on this?

**Comment by gazpachoking** [comment]
We need a champion willing to experiment and make a pull request

**Comment by BrutuZ** [comment]
Or for people to realize IMDB is hostile to third-party applications that aren't willing to pay their hefty API fees and should consider moving to friendlier services.
Trakt even added a import feature on their welcome wizard recently to ease such transitions https://releasenotes.trakt.tv/release/9DJ46-november-5-2024

**Comment by dj-nitehawk** [comment]
just threw together a small utlitity [web app](https://github.com/dj-nitehawk/ImdbWatchlistManager) to scrape the watchlist page and convert the data to csv, so i can use it with the csv plugin in flexget.

would've contributed to flexget, but can't remember the last time i touched python. the app is written in c#, but it might help someone to port it to flexget imdb plugin as c# is pretty easy to read/understand.

main parsing logic is [here](https://github.com/dj-nitehawk/ImdbWatchlistManager).

**Comment by ranirahn** [comment]
> main parsing logic is [here](https://github.com/dj-nitehawk/ImdbWatchlistManager/blob/116a983a4df02dff8336b3439ca60c582cfee849/Source/Features/Watchlist/Export/Endpoint.cs#L19-L60).

I have not tested it yet, but it seems like good alternative. I am not good at coding so my question is do it take all pages on watchlist or only first page? It gives 250 titles per page, but as I am so bad at coding I could not find do this code loop through all pages.
Right now I download csv file manually and upload to server, but if I can make this automatic with this it almost solves the issue for me.
